src/v/archival/ntp_archiver_service.h

```diff
@@ -655,6 +709,14 @@ class ntp_archiver {
     // be triggered by notify_leadership.
     ss::condition_variable _leader_cond;
 
+    // Used by clients wait()'ing on flush. Will be signalled by a complete
+    // flush(), or by a change in leadership.
+    ss::condition_variable _flush_cond;
+
+    // Indicates that a request to flush all local data up to offset() has been
```

Comment:
> up to offset()

can you make it explicit if this is inclusive or non-inclusive?

Reply:
Expanded comment to explicitly call out the fact `_flush_uploads_offset` is inclusive.